core/src/test/java/kafka/security/minikdc/MiniKdc.java

```diff
@@ -0,0 +1,512 @@
+        DEFAULT_CONFIG.put(TRANSPORT, "TCP");
+        DEFAULT_CONFIG.put(DEBUG, "false");
+    }
+
+    private final String orgName;
+    private final String orgDomain;
+    private final String realm;
+    private final String host;
+    private int port;
+
+    private final Properties config;
+    private final File workDir;
+    private final File krb5conf;
+
+    private DirectoryService ds;
+    private KdcServer kdc;
+    private boolean closed = false;
+
+    public MiniKdc(Properties config, File workDir) {
+        Set<String> requiredProperties = new HashSet<>(Arrays.asList(ORG_NAME, ORG_DOMAIN, KDC_BIND_ADDRESS, KDC_PORT,
+                INSTANCE, TRANSPORT, MAX_TICKET_LIFETIME, MAX_RENEWABLE_LIFETIME));
+        if (!config.keySet().containsAll(requiredProperties)) {
+            throw new IllegalArgumentException("Missing required properties: " + requiredProperties);
+        }
+        this.config = config;
+        this.workDir = workDir;
+
+        this.orgName = config.getProperty(ORG_NAME);
+        this.orgDomain = config.getProperty(ORG_DOMAIN);
+        this.realm = this.orgName.toUpperCase(Locale.ENGLISH) + "." + this.orgDomain.toUpperCase(Locale.ENGLISH);
+        this.krb5conf = new File(workDir, "krb5.conf");
+        this.host = config.getProperty(KDC_BIND_ADDRESS);
+        this.port = Integer.parseInt(config.getProperty(KDC_PORT));
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length < 4) {
+            System.out.println("Arguments: <WORKDIR> <MINIKDCPROPERTIES> <KEYTABFILE> [<PRINCIPALS>]+");
+            Exit.exit(1);
+        }
+
+        String workDirPath = args[0];
+        String configPath = args[1];
+        String keytabPath = args[2];
+        String[] principals = new String[args.length - 3];
+        System.arraycopy(args, 3, principals, 0, principals.length);
+
+        File workDir = new File(workDirPath);
+        if (!workDir.exists()) {
+            throw new RuntimeException("Specified work directory does not exist: " + workDir.getAbsolutePath());
+        }
+
+        Properties config = createConfig();
+        File configFile = new File(configPath);
+        if (!configFile.exists()) {
+            throw new RuntimeException("Specified configuration does not exist: " + configFile.getAbsolutePath());
+        }
+
+        Properties userConfig = Utils.loadProps(configFile.getAbsolutePath());
+        config.putAll(userConfig);
+        File keytabFile = new File(keytabPath).getAbsoluteFile();
+
+        start(workDir, config, keytabFile, Arrays.asList(principals));
+    }
+
+    /**
+     * Convenience method that returns MiniKdc default configuration.
+     *
+     * The returned configuration is a copy, it can be customized before using
+     * it to create a MiniKdc.
+     */
+    public static Properties createConfig() {
+        Properties properties = new Properties();
+        DEFAULT_CONFIG.forEach(properties::putIfAbsent);
```

Comment:
It seems `DEFAULT_CONFIG` is used by `createConfig` only. Could you please refactor it? for example:
```java
        Properties properties = new Properties();
        properties.put(ORG_NAME, "EXAMPLE");
        properties.put(ORG_DOMAIN, "COM");
        properties.put(KDC_BIND_ADDRESS, "localhost");
        properties.put(KDC_PORT, "0");
        properties.put(INSTANCE, "DefaultKrbServer");
        properties.put(MAX_TICKET_LIFETIME, "86400000");
        properties.put(MAX_RENEWABLE_LIFETIME, "604800000");
        properties.put(TRANSPORT, "TCP");
        properties.put(DEBUG, "false");
        return properties;
```